includes/src/Report/Accordians.js

```diff
@@ -3,18 +3,32 @@ import Collapse from './Collapse'
 
 class Accordians extends React.Component {
 
+    constructor(props) {
+        super(props);
+
+        this.getRank = this.getRank.bind(this);
+    };
+
     getRank(pluginDataList) {
         var testCaseMax = 0;
         var maxUserRank = -1;
         var maxOWTFRank = -1;
+        var selectedType = this.context.selectedType;
+        var selectedRank = this.context.selectedRank;
+        var selectedGroup = this.context.selectedGroup;
+        var selectedOwtfRank = this.context.selectedOwtfRank;
+        var selectedStatus = this.context.selectedStatus;
+
         for (var i = 0; i < pluginDataList.length; i++) {
-            if (pluginDataList[i].user_rank != null && pluginDataList[i].user_rank != -1) {
-                if (pluginDataList[i].user_rank > maxUserRank) {
-                    maxUserRank = pluginDataList[i].user_rank;
-                }
-            } else if (pluginDataList[i].owtf_rank != null && pluginDataList[i].owtf_rank != -1) {
-                if (pluginDataList[i].owtf_rank > maxOWTFRank) {
-                    maxOWTFRank = pluginDataList[i].owtf_rank;
+            if ((selectedType.length === 0 || selectedType.indexOf(pluginDataList[i].plugin_type) !== -1) && (selectedGroup.length === 0 || selectedGroup.indexOf(pluginDataList[i].plugin_group) !== -1) && (selectedRank.length === 0 || selectedRank.indexOf(pluginDataList[i].user_rank) !== -1) && (selectedOwtfRank.length === 0 || selectedOwtfRank.indexOf(pluginDataList[i].owtf_rank) !== -1) && (selectedStatus.length === 0 || selectedStatus.indexOf(pluginDataList[i].status) !== -1)) {
```

Comment:
Be consistent when matching conditions:
`indexOf(pluginDataList[i].plugin_type) !== -1)`  and `pluginDataList[i].owtf_rank != -1`